src/include/containers/Map.h

```diff
@@ -0,0 +1,173 @@
+#ifndef CUHKSZ_CONTAINERS_MAP
+#define CUHKSZ_CONTAINERS_MAP
+
+#include <map>
+#include <utility>
+
+namespace cuhksz {
+
+template <typename KeyType, typename ValueType>
+class Map {
+public:
+	typedef std::pair<const KeyType, ValueType> StdValueType;
+	Map();
+	Map(const Map& other);
+	Map( std::initializer_list<StdValueType> init );
+
+	~Map();
+
+	Map& operator =(Map& map2);
+
+	ValueType& get(const KeyType& key);
+	const ValueType& get(const KeyType& key) const;
```

Comment:
Maybe add `set` is better. `get` method can set value, but it is not obvious. Also operator `[ ]` can set value, but it will not do keychecking, I'll add a `set` which with keycheck.